### ISSUE TITLE
spooler body missing sporadically

### ISSUE THREAD

**Issue by fpn** [unbit/uwsgi#860]
Hi,

we see the body arg missing sporadically. Here is a quick test I wrote:
````
def spool_test():
    for i in range(1, 100000):
        body = {"a": 1}
        body = json.dumps(body)
        test_store.spool(body=body)

import uwsgi
@spool
def test_store(args):
    body = {}
    try:
        body = json.loads(args['body'])
    except:
        log.error("problem getting body from args=%s", args, exc_info=True)
        return uwsgi.SPOOL_OK
    print "."
    return uwsgi.SPOOL_OK
````

And then I get to see some  errors (e.g. 7 out 100000) with:
````
problem getting body from args={'ud_spool_func': 'test_store', 'spooler_task_name': 'uwsgi_spoolfile_on_alpha_14669_41408_0_1425513419_298061', 'ud_spool_ret': '-2'}
Traceback (most recent call last):
  File "....virtualenv/local/lib/python2.7/site-packages/uwsgidecorators.py", line 60, in manage_spool_request
    ret = f(vars)
 File "foo/test.py", line 904, in test_store
    body = json.loads(args['body'])
KeyError: 'body'
````
Tested with:
uwsgi==2.0.9
uwsgi==2.0.7

And this happens on both on Ubuntu 12.04 and 14.04 (python 2.7)

**Comment by unbit** [comment]
It would be interesting/useful to know if the problem happens when the spool file is generated or read.

Can you enforce SPOOL_RETRY in your test and check if the size of the spoolfiles is always the same ?

If not, can you attach one of the broken files here ?

Thanks

**Comment by funkybob** [comment]
Could this have been the same spool issue I ran into, which was fixed in 2.0.11?

**Comment by xrmx** [comment]
No user response since months and good chances this was the same @funkybob issue. Please reopen if it is sitll an issue.
